datafusion/optimizer/src/eliminate_duplicated_expr.rs

```diff
@@ -35,78 +36,111 @@ impl EliminateDuplicatedExpr {
         Self {}
     }
 }
-
+// use this structure to avoid initial clone
+#[derive(Eq, Clone, Debug)]
+struct SortExprWrapper {
```

Comment:
Wrap the Expr in a Wrapper to support specialized comparison

Reply:
This is very clever 👏